src/fu-plugin.c

```diff
@@ -967,6 +967,72 @@ fu_plugin_runner_update_cleanup (FuPlugin *plugin, FuDevice *device, GError **er
 	return TRUE;
 }
 
+gboolean
+fu_plugin_runner_update_attach (FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	FuPluginPrivate *priv = GET_PRIVATE (plugin);
+	FuPluginDeviceFunc func = NULL;
+
+	/* not enabled */
+	if (!priv->enabled)
+		return TRUE;
+
+	/* optional */
+	g_module_symbol (priv->module, "fu_plugin_update_attach", (gpointer *) &func);
+	if (func == NULL)
+		return TRUE;
+	g_debug ("performing update_attach() on %s", priv->name);
+	if (!func (plugin, device, error)) {
+		g_prefix_error (error, "failed to attach device %s: ", priv->name);
+		return FALSE;
+	}
+	return TRUE;
```

Comment:
This is a lot of boiler plate.  I almost wonder if it's better  to have a separate function
`fu_plugin_runner_call_vfunc `that does all of this but takes a 4th `const gchar *` argument for the vfunc being looked up.  The g_debug call can just use that same string as can the error.

Then this function turns into a one liner that looks like:
`return fu_plugin_runner_call_vfunc (plugin, device, error, "fu_plugin_update_attach");`

Reply:
Agreed, fixed.